spec/ts/noCF/smoke_spec.ts

```diff
@@ -0,0 +1,154 @@
+// Based off of spec/basic/elements_spec.js
+import {promise as ppromise, browser, element, by, By, $, $$, ExpectedConditions, ElementFinder} from '../../..';
+
+describe('ElementFinder', function() {
+  it('should return the same result as browser.findElement', async function() {
+    await browser.get('index.html#/form');
+    const nameByElement = element(by.binding('username'));
+
+    await expect(nameByElement.getText()).toEqual(
+        browser.findElement(by.binding('username')).getText());
+  });
+
+  it('should wait to grab the WebElement until a method is called', async function() {
+    // These should throw no error before a page is loaded.
+    const usernameInput = element(by.model('username'));
+    const name = element(by.binding('username'));
+
+    await browser.get('index.html#/form');
+
+    await expect(name.getText()).toEqual('Anon');
+
+    await usernameInput.clear();
+    await usernameInput.sendKeys('Jane');
+    await expect(name.getText()).toEqual('Jane');
+  });
+
+  it('should chain element actions', async function() {
+    await browser.get('index.html#/form');
+
+    const usernameInput = element(by.model('username'));
+    const name = element(by.binding('username'));
+
+    await expect(name.getText()).toEqual('Anon');
+
+    await ((usernameInput.clear() as any) as ElementFinder).sendKeys('Jane');
+    await expect(name.getText()).toEqual('Jane');
+  });
+
+  it('chained call should wait to grab the WebElement until a method is called',
+      async function() {
+    // These should throw no error before a page is loaded.
+    const reused = element(by.id('baz')).
+        element(by.binding('item.reusedBinding'));
+
+    await browser.get('index.html#/conflict');
+
+    await expect(reused.getText()).toEqual('Inner: inner');
+    await expect(reused.isPresent()).toBe(true);
+  });
+
+  it('should differentiate elements with the same binding by chaining',
+      async function() {
+        await browser.get('index.html#/conflict');
+
+        const outerReused = element(by.binding('item.reusedBinding'));
+        const innerReused =
+            element(by.id('baz')).element(by.binding('item.reusedBinding'));
+
+        await expect(outerReused.getText()).toEqual('Outer: outer');
+        await expect(innerReused.getText()).toEqual('Inner: inner');
+      });
+
+  it('should chain deeper than 2', async function() {
+    // These should throw no error before a page is loaded.
+    const reused = element(by.css('body')).element(by.id('baz')).
+        element(by.binding('item.reusedBinding'));
+
+    await browser.get('index.html#/conflict');
+
+    await expect(reused.getText()).toEqual('Inner: inner');
+  });
+
+  it('should allow handling errors', async function() {
+    await browser.get('index.html#/form');
+    await $('.nopenopenope').getText().then(async function(/* string */) {
```

Comment:
Would it make sense for this test to instead use try/catch (that works with async, yeah?)